client/templates/userprofile/settings.js

```diff
@@ -1,151 +1,113 @@
-var isEmail = function(email) {
-    var re = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
-    return re.test(email);
-};
-
-Template.profileSettings.rendered = function() {
-    $("#language").find("option[value=" + Meteor.user().language + "]").attr("selected", "selected");
-    $("#gender").find("option[value=" + Meteor.user().gender + "]").attr("selected", "selected");
-    Session.set("basicsuccess", "");
-    Session.set("passwordsuccess", "");
-};
-
-Template.profileSettings.helpers({
-    //TODO: fetch current profile
-    'profile': function() {
-        return Meteor.users.findOne({'id':this._id });
-    },
-    'usermail': function() {
-        return Meteor.user().emails[0].address;
-    },
-    'realerror': function() {
-        return Session.get("realerror");
-    },
-    'usererror': function() {
-        return Session.get("usererror");
-    },
-    'emailerror': function() {
-        return Session.get("emailerror");
-    },
-    'basicsuccess': function() {
-        return Session.get("basicsuccess");
-    },
-    'passwordsuccess': function() {
-        return Session.get("passwordsuccess");
-    },
-
-    'perrors': function() {
-        return Session.get("perrors");
-    },
-
-    'langs': function () {
-        return data = ["English", "Finnish", "German", "Persian", "Russian", "Swedish"];
-    }
-});
-
-Template.profileSettings.events({
-
-    'change #realName': function(event,template) {
-      if(template.find("#realName").value)
-      {
-          Session.set("realerror", "");
-      }
-        else
-      {
-          Session.set("realerror", "Your Realname is empty!");
-      }
-    },
-    'change #userName': function(event,template) {
-        if(template.find("#userName").value)
-        {
-            Session.set("usererror", "");
-        }
-        else
-        {
-            Session.set("usererror", "Your username is empty!");
-        }
-    },
-    'change #email': function(event,template) {
-        if(template.find("#email").value)
-        {
-            Session.set("emailerror", "");
-            if(!(isEmail(template.find("#email".value))))
-            {
-                Session.set("emailerror", "You have not entered a correct Email!");
-            }
-            else
-            {
-                Session.set("emailerror", "");
-            }
-        }
-        else
-        {
-            Session.set("emailerror", "Your Email is empty!");
+AutoForm.hooks({
```

Comment:
Create a separate `autoForm.js` file in the same folder as `settings.js`. In the `autoForm.js` file, add something to this effect:

```JavaScript
var userProfileHooks = {
  // Called when any submit operation succeeds
  onSuccess: function(formType, result) {
    return sAlert.success('Profile updated');
  }
};

AutoForm.addHooks('UserProfileEdit', userProfileHooks);
```